---
title: "DESeq2"
output: html_document
---
Rmd to use `DESeq2` package on 4 libraries count matrix. Differential gene expression between infection statuses, then influence of temperature on the expression of those differentially expressed genes. 

Library info --> all from sampling day 2         
380822_cold_uninfected      
380823_cold_infected       
380824_warm_uninfected      
380825_warm_infected      

Load packages:    
```{r}
library(DESeq2)
library(tidyverse)
library(pheatmap)
library(RColorBrewer)
library(data.table)
```


### Read in count matrix for the 4 libraries from transcriptome v 3.1
file created using https://github.com/RobertsLab/paper-tanner-crab/blob/master/notebooks/kallisto-4libraries.ipynb 
```{r}
countmatrix <- read.delim("../analyses/kallisto-0812/kallisto-0812.isoform.counts.matrix", header = TRUE, sep = '\t')
rownames(countmatrix) <- countmatrix$X
countmatrix <- countmatrix[,-1]
head(countmatrix)
```

Rename columns just to remove the "X" before each sample name:
```{r}
colnames(countmatrix) <- c("380822_cold_uninfected", "380823_cold_infected", "380824_warm_uninfected", "380825_warm_infected")
head(countmatrix)
```

Round integers up to hole numbers for further analysis:
```{r}
countmatrix <- round(countmatrix, 0)
head(countmatrix)
```

## Get DEGs based on infection 
Not including temperature as part of the design      
```{r}
colData <- data.frame(condition=factor(c("uninfected", "infected", "uninfected", "infected")), 
                             type=factor(rep("paired-end", 4)))
rownames(colData) <- colnames(countmatrix)
dds <- DESeqDataSetFromMatrix(countData = countmatrix,
                                     colData = colData, 
                                     design = ~ condition)
```

```{r}
dds <- DESeq(dds)
res <- results(dds)
res <- res[order(rownames(res)), ]
```

```{r}
head(res)
```

```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(res[!is.na(res$padj) & res$padj <= 0.05, ])
```
1343 rows! DEGs just based on infection status (irrespective of temperature)

```{r}
infection_fig <- res
# The main plot
plot(infection_fig$baseMean, infection_fig$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     main="Infection Status  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
infection_fig.sig <- res[!is.na(res$padj) & res$padj <= 0.05, ]
points(infection_fig.sig$baseMean, infection_fig.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```

```{r}
#write.table(infection_fig.sig, "../analyses/DESeq2/DEGlist-infectionONLY.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
Wrote out file 08/12/2020. Commented out code. 

## Get influence of temperature on the DEGs from above
Taking temperature into account in the design.          
Code based on the Multi-factor design section of the manual:        http://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#multi-factor-designs 

```{r}
deseq2.colData <- data.frame(condition=factor(c("uninfected", "infected", "uninfected", "infected")), 
                             type=factor(rep("paired-end", 4)),
                             temperature=factor(c("decreased", "decreased", "elevated", "elevated")))
rownames(deseq2.colData) <- colnames(countmatrix)
deseq2.dds <- DESeqDataSetFromMatrix(countData = countmatrix,
                                     colData = deseq2.colData, 
                                     design = ~ condition + temperature)
```

Check levels of temperature and condition (infection status)
```{r}
levels(deseq2.dds$temperature)
```

```{r}
levels(deseq2.dds$condition)
```

`DESeq2` automatically puts the levels in alphabetical order and the first listed level is the reference level for the factor. 

We want uninfected to be the reference. 
```{r}
deseq2.dds$condition = relevel(deseq2.dds$condition, "uninfected")
levels(deseq2.dds$condition)
```

The following will pull the results from `condition` (infection status) because that is our variable of interest. This tells us how temperature contributes to the infection DEGs
```{r}
design(deseq2.dds) <- formula(~ temperature + condition)
deseq2.dds <- DESeq(deseq2.dds)
```

Access results:
```{r}
deseq2.res <- results(deseq2.dds)
head(deseq2.res)
```

```{r}
summary(deseq2.res)
```

```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(deseq2.res[!is.na(deseq2.res$padj) & deseq2.res$padj <= 0.05, ])
```
408 DEGs between infection statuses that are influence by temperature

```{r}
inf <- deseq2.res
# The main plot
plot(inf$baseMean, inf$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     #main="Infection Status  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
inf.sig <- deseq2.res[!is.na(deseq2.res$padj) & deseq2.res$padj <= 0.05, ]
points(inf.sig$baseMean, inf.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```

```{r}
#write.table(inf.sig, "../analyses/DESeq2/DEGlist-infection-with-temp.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
Wrote out table 08/13/2020. Commented out code

## Now to figure out what these results mean: 
In the multifactor section of the `DESeq2` manual:                
The contrast argument of the function _results_ needs a character vector of three componenets: the name of the variable (in this case "temperature"), and the name of the factor level for the numerator of the log2 ratio (elevated) and the denominator (decreased) 

A **contrast** is a linear combination of estimated log2 fold changes. Can be used to test if differences between groups are equal to zero.         
```{r}
resultsNames(deseq2.dds)
```

```{r}
deseq2.resTemp <- results(deseq2.dds,
                          contrast = c("temperature", "elevated",  "decreased"))
head(deseq2.resTemp)
```

```{r}
tmp <- deseq2.resTemp
# The main plot
plot(tmp$baseMean, tmp$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     #main="Infection Status  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
tmp.sig <- deseq2.resTemp[!is.na(deseq2.resTemp$padj) & deseq2.resTemp$padj <= 0.05, ]
points(tmp.sig$baseMean, tmp.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```

```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(deseq2.resTemp[!is.na(deseq2.resTemp$padj) & deseq2.resTemp$padj <= 0.05, ])
```
123 DEGs - genes that are influenced by temperature that are associated with infection status

```{r}
summary(deseq2.resTemp)
```

```{r}
#write.table(tmp.sig, "../analyses/DESeq2/DEGlist-contrast_temperature.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
Wrote out 08/12/2020. Comment out code. 

## `join` count matrix with lists of DEGs

Make rownames into a column called "Trinity_ID" for the count matrix:
```{r}
countmatrix <- tibble::rownames_to_column(countmatrix, "Trinity_ID")
head(countmatrix)
```


### Read in list of 1343 DEGs:
```{r}
degs_infONLY <- read.delim("../analyses/DESeq2/DEGlist-infectionONLY.tab", sep = '\t')
head(degs_infONLY)
```

Make rownames into a column called "Trinity_ID":
```{r}
degs_infONLY <- tibble::rownames_to_column(degs_infONLY, "Trinity_ID")
head(degs_infONLY)
```

`join` count matrix with degs_infONLY:
```{r}
degs_infONLY_counts <- left_join(countmatrix, degs_infONLY, by = "Trinity_ID")
head(degs_infONLY_counts)
```

Only want the ones that match!
```{r}
degs_infONLY_counts_match <- filter(degs_infONLY_counts, baseMean != "NA")
head(degs_infONLY_counts_match)
```
1343 rows. Perfect! 

Write out list with count data:
```{r}
#write.table(degs_infONLY_counts_match, "../analyses/DESeq2/DEGlist-infectionONLY-counts.tab", sep = '\t', quote = FALSE, row.names = FALSE)
```
wrote out 08/13/2020. Comment out code. 

### Read in list of 183 DEGs:
```{r}
degs_inftemp <- read.delim("../analyses/DESeq2/DEGlist-infection_influenced-by-temp.tab", sep = '\t')
head(degs_inftemp)
```

Make rownames into a column called "Trinity_ID":
```{r}
degs_inftemp <- tibble::rownames_to_column(degs_inftemp, "Trinity_ID")
head(degs_inftemp)
```

`join` count matrix with degs_inftemp:
```{r}
degs_inftemp_counts <- left_join(countmatrix, degs_inftemp, by = "Trinity_ID")
head(degs_inftemp_counts)
```

Only want the ones that match!
```{r}
degs_inftemp_counts_match <- filter(degs_inftemp_counts, baseMean != "NA")
head(degs_inftemp_counts_match)
```

Write out list with count data:
```{r}
#write.table(degs_inftemp_counts_match, "../analyses/DESeq2/DEGlist-infection-with-temp-counts.tab", sep = '\t', quote = FALSE, row.names = FALSE)
```
Wrote out 08/13/2020. Commented out code. 

### Read in list of 123 DEGs:
DEG list of genes influenced by temperature associated with infection status comparison.   

```{r}
degs_contrasttemp <- read.delim("../analyses/DESeq2/DEGlist-contrast_temperature.tab", sep = '\t')
head(degs_contrasttemp)
```

Make rownames into a column called "Trinity_ID":
```{r}
degs_contrasttemp <- tibble::rownames_to_column(degs_contrasttemp, "Trinity_ID")
head(degs_contrasttemp)
```

`join` count matrix with degs_inftemp:
```{r}
degs_contrasttemp_counts <- left_join(countmatrix, degs_contrasttemp, by = "Trinity_ID")
head(degs_contrasttemp_counts)
```

Only want the ones that match!
```{r}
degs_contrasttemp_counts_match <- filter(degs_contrasttemp_counts, baseMean != "NA")
head(degs_contrasttemp_counts_match)
```
123 rows! Good. 

Write out list with count data:
```{r}
#write.table(degs_contrasttemp_counts_match, "../analyses/DESeq2/DEGlist-contrast_temperature-counts.tab", sep = '\t', quote = FALSE, row.names = FALSE)
```
wrote out 08/13/2020. commented out code. 

