---
title: "DESeq2"
output: html_document
---
Rmd to use `DESeq2` package on 4 libraries count matrix. Differential gene expression between infection statuses, then influence of temperature on the expression of those differentially expressed genes. 

Library info --> all from sampling day 2         
380822_cold_uninfected      
380823_cold_infected       
380824_warm_uninfected      
380825_warm_infected      

Load packages:    
```{r}
library(DESeq2)
library(tidyverse)
library(pheatmap)
library(RColorBrewer)
```


### Read in count matrix for the 4 libraries from transcriptome v 3.1
file created using https://github.com/RobertsLab/paper-tanner-crab/blob/master/notebooks/kallisto-4libraries.ipynb 
```{r}
countmatrix <- read.delim("../analyses/kallisto-0812/kallisto-0812.isoform.counts.matrix", header = TRUE, sep = '\t')
rownames(countmatrix) <- countmatrix$X
countmatrix <- countmatrix[,-1]
head(countmatrix)
```

Rename columns just to remove the "X" before each sample name:
```{r}
colnames(countmatrix) <- c("380822_cold_uninfected", "380823_cold_infected", "380824_warm_uninfected", "380825_warm_infected")
head(countmatrix)
```

Round integers up to hole numbers for further analysis:
```{r}
countmatrix <- round(countmatrix, 0)
head(countmatrix)
```

## Get DEGs based on infection 
Not including temperature as part of the design      
```{r}
colData <- data.frame(condition=factor(c("uninfected", "infected", "uninfected", "infected")), 
                             type=factor(rep("paired-end", 4)))
rownames(colData) <- colnames(countmatrix)
dds <- DESeqDataSetFromMatrix(countData = countmatrix,
                                     colData = colData, 
                                     design = ~ condition)
```

```{r}
dds <- DESeq(dds)
res <- results(dds)
res <- res[order(rownames(res)), ]
```

```{r}
head(res)
```

```{r}
# Count number of hits with adjusted p-value less then 0.05
dim(res[!is.na(res$padj) & res$padj <= 0.05, ])
```
1343 rows! DEGs just based on infection status (irrespective of temperature)

```{r}
infection_fig <- res
# The main plot
plot(infection_fig$baseMean, infection_fig$log2FoldChange, pch=20, cex=0.45, ylim=c(-15, 15), log="x", col="darkgray",
     main="Infection Status  (pval </= 0.05)",
     xlab="mean of normalized counts",
     ylab="Log2 Fold Change")
# Getting the significant points and plotting them again so they're a different color
infection_fig.sig <- res[!is.na(res$padj) & res$padj <= 0.05, ]
points(infection_fig.sig$baseMean, infection_fig.sig$log2FoldChange, pch=20, cex=0.45, col="red")
# 2 FC lines
abline(h=c(-1,1), col="blue")
```

```{r}
write.table(infection_fig.sig, "../analyses/DESeq2/DEGlist-infectionONLY.tab", sep = "\t", row.names = T, quote = FALSE, col.names = TRUE)
```
Wrote out file 08/12/2020. Commented out code. 




