---
title: "heatmaps-singlecrab_over-time"
output: html_document
---
Rmd to make heatmaps using `pheatmap` of gene over time in an individual crab. This crab was infected with _Hematodinium_, kept in ambient throughout the experiment, and hemolymph was sampled on day 0, day 2, and day 17 (final day of experiment). 

I will also annotate gene list with the `blast` output from transcriptome v 3.1 and uniprot/swissprot and gene ontology. 

Gene count matrix is from transcriptome v 3.1. 

### Load packages
```{r}
library(dplyr)
library(tidyverse)
library(pheatmap)
library(data.table)
library(RColorBrewer)
```

## Read in files: 

1. `blast` output from transcriptome v 3.1 with uniprot and GO:
```{r}
blast_go <- read.delim("../analyses/BLAST-to-GOslim/transcriptomev3.1-blast-uniprotGO.tab", header = TRUE)
head(blast_go)
```

### Load in data of counts from individual crab RNAseq (transcriptome v 3.1): 

ambient, infected crab. days 0, 2, and 17
```{r}
norm_counts <- read.delim("../analyses/kallisto-single_crab_over_time/kallisto-single-crab.isoform.counts.matrix")
head(norm_counts)
```

Set Trinity_ID as row names:
```{r}
rownames(norm_counts) <- norm_counts$X #set Trinity IDs as rownames
norm_counts <- norm_counts[,-1] #remove redundant column
head(norm_counts)
```

ROUND INTEGERS UP:
```{r}
norm_counts <- round(norm_counts,0)
head(norm_counts)
```

Set rownames as a column again:     
```{r}
norm_counts <- tibble::rownames_to_column(norm_counts, "Trinity_ID")
head(norm_counts)
```

Rename sample columns:     
Library IDs: 178 for Day 0; 359 for Day 2, and 463 for Dday 17
```{r}
colnames(norm_counts) <- c("Trinity_ID", "Day0", "Day02", "Day17")
head(norm_counts)
```

Set heatmap colors:
```{r}
heatmapBrBG <- RColorBrewer::brewer.pal(11, "BrBG")
```

### Make heatmap using `pheatmap`
78649 rows      
Filter out rows that have 0 across all samples:
```{r}
norm_counts <- norm_counts[rowSums(norm_counts[, -1] >0) !=0, ]
head(norm_counts)
```
Now 39158 rows.    

Now lets remove low-count genes across all samples.        
Sum all rows in new column:            
```{r}
norm_counts$sums <- rowSums( norm_counts[,2:4] )
head(norm_counts)
```

Remove columns with sums <= 64. This number (64) was chosen because it is the smallest number that can still make a heatmap. 
```{r}
norm_counts_highcts <- filter(norm_counts, sums >= 64)
head(norm_counts_highcts)
```
Now there's 13,954 rows. 

Remove "sums" column from count matrix for heatmap:      
```{r}
crab.heatmap <- select(norm_counts_highcts, "Trinity_ID", "Day0", "Day02", "Day17")
head(crab.heatmap)
```

Have to make trinity ID column into rownames:
Set Trinity_ID as row names:
```{r}
rownames(crab.heatmap) <- crab.heatmap$Trinity_ID #set Trinity IDs as rownames
crab.heatmap <- crab.heatmap[,-1] #remove redundant column
head(crab.heatmap)
```

```{r}
pheatmap(crab.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapBrBG, fontsize_col = 12, fontsize_row = 12)
```

### Now annotate heatmap with clustering and sample info. 

Set heatmap as an object:
```{r}
crab.over.time <- pheatmap(crab.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapBrBG, fontsize_col = 12, fontsize_row = 12)
```

Extract 6 clusters:   
Chose 6 because when I chose 4, it clumped some of the bigger clusters together.
```{r}
clust <- cbind(crab.heatmap,
                   cluster = cutree(crab.over.time$tree_row,
                                     k = 6))
head(clust)
```

Pull out just the column of cluster numbers:
```{r}
clusters <- select(clust, "cluster")
head(clusters)
```

Rename the clusters in the rows to "cluster #" instead of just the number. 
```{r}
clusters <- data.frame(cluster = ifelse(clusters == 1, "cluster1", ifelse(clusters == 2, "cluster2", ifelse(clusters == 3, "cluster3", ifelse(clusters == 4, "cluster4", ifelse(clusters == 5, "cluster5", "cluster6"))))))
head(clusters)
```

The following four chunks are commented out because they are now unnecessary/redundant

Add column annotations for the samples denoting temperature treatment and infection status, and sampling day:
```{r}
#sample_col <- data.frame(temperature = c("ambient", "ambient", "ambient"))
#row.names(sample_col) <-colnames(crab.heatmap)
#sample_col
```

add infection status info:
```{r}
#sample_col$infection.status <- c("infected", "infected", "infected")
#sample_col
```

add sampling day info:
```{r}
#sample_col$sampling.day <- c("day0", "day2", "day17")
#sample_col
```

specify colors for annotation columns:
```{r}
my_colors <- list(
  cluster = c(cluster1 = "#1B9E77", cluster2 = "#D95F02", cluster3 = "#7570B3", cluster4 = "#E7298A", cluster5 = "#cc4ee0", cluster6 = "#82ed82")
)
my_colors
```


```{r}
pheatmap(crab.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", annotation_colors = my_colors, annotation_row = clusters, show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapBrBG, fontsize_col = 12, fontsize_row = 12)
```
Saved heatmap by right clicking > Save Image As, download as png, then open in preview, export > PDF, save. 

save heatmap to analyses/pheatmap:
```{r}
#pdf("../analyses/pheatmap/individual-crab-over-time.pdf", width = 8, height = 8, )
#pheatmap(crab.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", annotation_colors = my_colors, annotation_row = clusters, show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapBrBG, fontsize_col = 12, fontsize_row = 12)
#dev.off()
```
Save pdf for publication. 2022-02-24

### `join` heatmap info with clusters, `blast` output from transcriptome v 3.1 and uniprot-SP-GO. save to analyses/heatmaps

Make heatmap matrix into a table with rows names as a column called "Trinity_ID":
```{r}
clust2 <- tibble::rownames_to_column(clust, "Trinity_ID")
head(clust2)
```

`join` with `blast` and uniprot GO output:
```{r}
crab.blast.go <- left_join(clust2, blast_go, by = "Trinity_ID")
head(crab.blast.go)
```

Write out into paper-tanner-crab/analyses/heatmaps:
```{r}
#write.table(crab.blast.go, "../analyses/pheatmap/single_crab-clusters-blast-GO.tab", sep = '\t', quote = FALSE, row.names = FALSE)
```
Wrote out 02/11/2021. Comment out code. 




