---
title: "heatmaps-singlecrab_over-time"
output: html_document
---
Rmd to make heatmaps using `pheatmap` of gene over time in an individual crab. This crab was infected with _Hematodinium_, kept in ambient throughout the experiment, and hemolymph was sampled on day 0, day 2, and day 17 (final day of experiment). 

I will also annotate gene list with the `blast` output from transcriptome v 3.1 and uniprot/swissprot and gene ontology. 

Gene count matrix is from transcriptome v 3.1. 

### Load packages
```{r}
library(dplyr)
library(tidyverse)
library(pheatmap)
library(data.table)
library(RColorBrewer)
```

## Read in files: 

1. `blast` output from transcriptome v 3.1:
```{r}
blast <- read.delim("../analyses/BLAST-to-GOslim/_blast-sep.tab", header = FALSE)
head(blast)
```

Rename first and third column for `join`-ing purposes later on. 
```{r}
colnames(blast) <- c("Trinity_ID", "V2", "uniprot_acc_ID", "V4", "V5", "V6", "V7", "V8", "V9", "V10", "V11", "V12", "V13", "V14")
head(blast)
```

2. Read in uniprot-SP-GO file
```{r}
uniprot_SP_GO <- read.delim("http://owl.fish.washington.edu/halfshell/bu-alanine-wd/17-07-20/uniprot-SP-GO.sorted", sep = '\t', header = FALSE)
head(uniprot_SP_GO)
```

Rename columns:
```{r}
colnames(uniprot_SP_GO) <- c("uniprot_acc_ID", "V2", "V3", "V4", "V5", "V6", "V7", "V8", "V9", "V10", "V11", "V12")
head(uniprot_SP_GO)
```

### Load in data of counts from individual crab RNAseq (transcriptome v 3.1):   
```{r}
norm_counts <- read.delim("../analyses/kallisto-single_crab_over_time/kallisto-single-crab.isoform.counts.matrix")
head(norm_counts)
```

Set Trinity_ID as row names:
```{r}
rownames(norm_counts) <- norm_counts$X #set Trinity IDs as rownames
norm_counts <- norm_counts[,-1] #remove redundant column
head(norm_counts)
```

ROUND INTEGERS UP:
```{r}
norm_counts <- round(norm_counts,0)
head(norm_counts)
```

Set rownames as a column again:     
```{r}
norm_counts <- tibble::rownames_to_column(norm_counts, "Trinity_ID")
head(norm_counts)
```

Rename sample columns: 
```{r}
colnames(norm_counts) <- c("Trinity_ID", "178", "359", "463")
head(norm_counts)
```

Set heatmap colors:
```{r}
heatmapBrBG <- RColorBrewer::brewer.pal(11, "BrBG")
```

### Make heatmap using `pheatmap`
78649 rows      
Filter out rows that have 0 across all samples:
```{r}
norm_counts <- norm_counts[rowSums(norm_counts[, -1] >0) !=0, ]
head(norm_counts)
```
Now 39158 rows.    

Now lets remove low-count genes across all samples.        
Sum all rows in new column:            
```{r}
norm_counts$sums <- rowSums( norm_counts[,2:4] )
head(norm_counts)
```

Remove columns with sums <= 64. This number (64) was chosen because it is the smallest number that can still make a heatmap. 
```{r}
norm_counts_highcts <- filter(norm_counts, sums >= 64)
head(norm_counts_highcts)
```
Now there's 13,954 rows. 

Remove "sums" column from count matrix for heatmap:      
```{r}
crab.heatmap <- select(norm_counts_highcts, "Trinity_ID", "178", "359", "463")
head(crab.heatmap)
```

Have to make trinity ID column into rownames:
Set Trinity_ID as row names:
```{r}
rownames(crab.heatmap) <- crab.heatmap$Trinity_ID #set Trinity IDs as rownames
crab.heatmap <- crab.heatmap[,-1] #remove redundant column
head(crab.heatmap)
```

```{r}
pheatmap(crab.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapBrBG, fontsize_col = 12, fontsize_row = 12)
```

### Now annotate heatmap with clustering and sample info. 

Set heatmap as an object:
```{r}
crab.over.time <- pheatmap(crab.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapBrBG, fontsize_col = 12, fontsize_row = 12)
```

Extract 6 clusters:
```{r}
clust <- cbind(crab.heatmap,
                   cluster = cutree(crab.over.time$tree_row,
                                     k = 6))
head(clust)
```

Pull out just the column of cluster numbers:
```{r}
clusters <- select(clust, "cluster")
head(clusters)
```

Rename the clusters in the rows to "cluster #" instead of just the number. 
```{r}
clusters <- data.frame(cluster = ifelse(clusters == 1, "cluster1", ifelse(clusters == 2, "cluster2", ifelse(clusters == 3, "cluster3", ifelse(clusters == 4, "cluster4", ifelse(clusters == 5, "cluster5", "cluster6"))))))
head(clusters)
```

Add column annotations for the samples denoting temperature treatment and infection status, and sampling day:
```{r}
sample_col <- data.frame(temperature = c("ambient", "ambient", "ambient"))
row.names(sample_col) <-colnames(crab.heatmap)
sample_col
```

add infection status info:
```{r}
sample_col$infection.status <- c("infected", "infected", "infected")
sample_col
```

add sampling day info:
```{r}
sample_col$sampling.day <- c("day0", "day2", "day17")
sample_col
```

specify colors for annotation columns:
```{r}
my_colors <- list(
  temperature = c(ambient = "orange"),
  infection.status = c(infected = "yellow"),
  sampling.day = c(day0 = "green", day2 = "red", day17 = "blue"),
  cluster = c(cluster1 = "#1B9E77", cluster2 = "#D95F02", cluster3 = "#7570B3", cluster4 = "#E7298A", cluster5 = "#cc4ee0", cluster6 = "#82ed82")
)
my_colors
```


```{r}
pheatmap(crab.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", annotation_colors = my_colors, annotation_row = clusters, annotation_col = sample_col, show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 200, legend = TRUE, color = heatmapBrBG, fontsize_col = 12, fontsize_row = 12)
```

save heatmap to analyses/pheatmap:
```{r}
pdf("../analyses/pheatmap/individual-crab-over-time.pdf", width = 11, height = 8.5)
pheatmap(crab.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", annotation_colors = my_colors, annotation_row = clusters, annotation_col = sample_col, show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapBrBG, fontsize_col = 12, fontsize_row = 12)
dev.off()
```

# Make heatmap of DEGs from `DESeq2` analyses in individual crab over time. 

Read in count matrix from crab over time - should be 78649 rows:
```{r}
norm_counts <- read.delim("../analyses/kallisto-single_crab_over_time/kallisto-single-crab.isoform.counts.matrix")
head(norm_counts)
```

Set Trinity_ID as row names:
```{r}
rownames(norm_counts) <- norm_counts$X #set Trinity IDs as rownames
norm_counts <- norm_counts[,-1] #remove redundant column
head(norm_counts)
```

ROUND INTEGERS UP:
```{r}
norm_counts <- round(norm_counts,0)
head(norm_counts)
```

Set rownames as a column again:     
```{r}
norm_counts <- tibble::rownames_to_column(norm_counts, "Trinity_ID")
head(norm_counts)
```


### Try with infection status comparison DEG list 
Read in DEG list from infection status comparison, not taking temperature into account in any way:
```{r}
infection_degs <- read.delim("../analyses/DESeq2/DEGlist-infectionONLY.tab", sep = '\t')
head(infection_degs)
```

Set rownames as a column called "Trinity_ID" 
```{r}
infection_degs <- tibble::rownames_to_column(infection_degs, "Trinity_ID")
head(infection_degs)
```

`join` the DEGs with the norm_counts matrix to find out how many match between the two tables:
```{r}
indiv_infection <- left_join(norm_counts, infection_degs, by = "Trinity_ID")
head(indiv_infection)
```

Just want the ones that match. Filter out just the rows that have something in column "". 
```{r}
indiv_infection <- filter(indiv_infection, baseMean != "NA")
head(indiv_infection)
```
It is now 1343 rows.

Pull out counts for the samples to make a heatmap
```{r}
infection.heatmap <- select(indiv_infection, "Trinity_ID", "X178_ambient_infected_0", "X359_ambient_infected_2", "X463_ambient_infected_17")
head(infection.heatmap)
```

Set Trinity IDs as rownames:
```{r}
rownames(infection.heatmap) <- infection.heatmap$Trinity_ID #set Trinity IDs as rownames
infection.heatmap <- infection.heatmap[,-1] #remove redundant column
head(infection.heatmap)
```

Rename columns:
```{r}
colnames(infection.heatmap) <- c(178, 359, 463)
head(infection.heatmap)
```

filter out rows that have zero counts across:
```{r}
infection.heatmap <- infection.heatmap[rowSums(infection.heatmap[, -1] >0) !=0, ]
head(infection.heatmap)
```
Now it's 551 rows. 

```{r}
pheatmap(infection.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapBrBG, fontsize_col = 12, fontsize_row = 12)
```

### Try with infection taking temperature into account list 
Read in DEGs (408) of infection status taking temperature into account:
```{r}
inftemp_degs <- read.delim("../analyses/DESeq2/DEGlist-infection-with-temp.tab", sep = '\t')
head(inftemp_degs)
```

Set rownames as a column called "Trinity_ID" 
```{r}
inftemp_degs <- tibble::rownames_to_column(inftemp_degs, "Trinity_ID")
head(inftemp_degs)
```

`join` the DEGs with the norm_counts matrix to find out how many match between the two tables:
```{r}
indiv_inftemp <- left_join(norm_counts, inftemp_degs, by = "Trinity_ID")
head(indiv_inftemp)
```

Just want the ones that match. Filter out just the rows that have something in column "". 
```{r}
indiv_inftemp <- filter(indiv_inftemp, baseMean != "NA")
head(indiv_inftemp)
```
408. 

Pull out counts for the samples to make a heatmap
```{r}
inftemp.heatmap <- select(indiv_inftemp, "Trinity_ID", "X178_ambient_infected_0", "X359_ambient_infected_2", "X463_ambient_infected_17")
head(inftemp.heatmap)
```

Set Trinity IDs as rownames:
```{r}
rownames(inftemp.heatmap) <- inftemp.heatmap$Trinity_ID #set Trinity IDs as rownames
inftemp.heatmap <- inftemp.heatmap[,-1] #remove redundant column
head(inftemp.heatmap)
```

Rename columns:
```{r}
colnames(inftemp.heatmap) <- c(178, 359, 463)
head(inftemp.heatmap)
```

filter out rows that have zero counts across:
```{r}
inftemp.heatmap <- inftemp.heatmap[rowSums(inftemp.heatmap[, -1] >0) !=0, ]
head(inftemp.heatmap)
```
Now 212 rows. 

```{r}
pheatmap(inftemp.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapBrBG, fontsize_col = 12, fontsize_row = 12)
```

### Try with infection DEGs that are influenced by temperature 
Read in DEG (123) list of infection status comparison that are influenced by temperature. 
```{r}
tempcontrast_degs <- read.delim("../analyses/DESeq2/DEGlist-contrast_temperature.tab", sep = '\t')
head(tempcontrast_degs)
```

Set rownames as a column called "Trinity_ID" 
```{r}
tempcontrast_degs <- tibble::rownames_to_column(tempcontrast_degs, "Trinity_ID")
head(tempcontrast_degs)
```

`join` the DEGs with the norm_counts matrix to find out how many match between the two tables:
```{r}
indiv_tempcon <- left_join(norm_counts, tempcontrast_degs, by = "Trinity_ID")
head(indiv_tempcon)
```

Just want the ones that match. Filter out just the rows that have something in column "". 
```{r}
indiv_tempcon <- filter(indiv_tempcon, baseMean != "NA")
head(indiv_tempcon)
```

Pull out counts for the samples to make a heatmap
```{r}
tempcon.heatmap <- select(indiv_tempcon, "Trinity_ID", "X178_ambient_infected_0", "X359_ambient_infected_2", "X463_ambient_infected_17")
head(tempcon.heatmap)
```

Set Trinity IDs as rownames:
```{r}
rownames(tempcon.heatmap) <- tempcon.heatmap$Trinity_ID #set Trinity IDs as rownames
tempcon.heatmap <- tempcon.heatmap[,-1] #remove redundant column
head(tempcon.heatmap)
```

Rename columns:
```{r}
colnames(tempcon.heatmap) <- c(178, 359, 463)
head(tempcon.heatmap)
```

filter out rows that have zero counts across:
```{r}
tempcon.heatmap <- tempcon.heatmap[rowSums(tempcon.heatmap[, -1] >0) !=0, ]
head(tempcon.heatmap)
```
94 rows. 

```{r}
pheatmap(tempcon.heatmap, scale = "row", cluster_row = TRUE, cluster_cols = FALSE, clustering_distance_rows = "euclidean", clustering_method = "average", show_rownames = FALSE, show_colnames = TRUE, treeheight_col = 80, legend = TRUE, color = heatmapBrBG, fontsize_col = 12, fontsize_row = 12)
```







